### PR TITLE
Add fix to handle tests outside tests directory

### LINKED ISSUE #20359: [Testerina] BAD SAD When tests are defined outside the tests directory.
**Description:**
When I define a test outside the test directory, and do ballerina test <module_name>, ballerina crashes.

**Steps to reproduce:**
**Main.bal**
import ballerina/test;

@test:Config{}
function testAssertIntEquals() {
int answer = 8;
test:assertEquals(answer, 8, msg = "int values not equal");
}

**Command**
ballerina test test_module

**LOG**
[2019-12-12 14:01:22,638] SEVERE {b7a.log.crash} - null
java.lang.NullPointerException
at org.ballerinalang.testerina.core.BTestRunner.lambda$null$26(BTestRunner.java:745)
at java.util.ArrayList.forEach(ArrayList.java:1257)
at org.ballerinalang.testerina.core.BTestRunner.lambda$execute$28(BTestRunner.java:673)
at java.lang.Iterable.forEach(Iterable.java:75)
at org.ballerinalang.testerina.core.BTestRunner.execute(BTestRunner.java:630)
at org.ballerinalang.testerina.core.BTestRunner.runTest(BTestRunner.java:160)
at org.ballerinalang.testerina.util.TesterinaUtils.executeTests(TesterinaUtils.java:132)
at org.ballerinalang.packerina.task.RunTestsTask.execute(RunTestsTask.java:90)
at org.ballerinalang.packerina.TaskExecutor.executeTasks(TaskExecutor.java:38)
at org.ballerinalang.packerina.cmd.TestCommand.execute(TestCommand.java:282)
at java.util.Optional.ifPresent(Optional.java:159)
at org.ballerinalang.tool.Main.main(Main.java:57)

**Affected Versions:**
ballerina -v
jBallerina 1.1.0-alpha
Language specification 2019R3
Command 1.1.0-alpha

**OS, DB, other environment details and versions:**
MacOSX(10.15.1)

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**
@Kishanthan

### PR DESCRIPTION
**Pull request by Dilhasha**
- When there are tests defined outside the tests directory, a Null pointer exception is thrown.
- When there is at least one test defined inside the tests directory, tests outside the tests directory are also executed. 

Fixes #20359

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/TestAnnotationProcessor.java [hunk 1]
```diff
@@ -81,14 +83,24 @@ public void process(FunctionNode functionNode, List<AnnotationAttachmentNode> an
...
+        String packageName = getPackageName(bLangPackage);
         suite = registry.getTestSuites().get(packageName);
         // Check if the registry contains a test suite for the package
         if (suite == null) {
-            // Add a test suite to the registry if it does not contain one pertaining to the package name
-            registry.getTestSuites().computeIfAbsent(packageName, func -> new TestSuite(packageName));
-            // Get the test suite related to the package from registry
-            suite = registry.getTestSuites().get(packageName);
+            //Set testable flag for single bal file execution
+            if ((Names.DOT.getValue()).equals(packageName)) {
```
Comment (warunalakshitha) [inline misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/TestAnnotationProcessor.java]: Are we checking Testable flag any other place? if not we can combine both if conditions

File: misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/TestAnnotationProcessor.java [hunk 2]
```diff
@@ -81,14 +83,24 @@ public void process(FunctionNode functionNode, List<AnnotationAttachmentNode> an
...
+            if (bLangPackage.getFlags().contains(Flag.TESTABLE)) {
+                // Add a test suite to the registry if it does not contain one pertaining to the package name
+                registry.getTestSuites().computeIfAbsent(packageName, func -> new TestSuite(packageName));
+                // Get the test suite related to the package from registry
+                suite = registry.getTestSuites().get(packageName);
+            } else {
+                return;
+            }
         }
         // Remove the duplicated annotations.
```
Comment (warunalakshitha) [inline misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/TestAnnotationProcessor.java]: Shall we write a test case for this

**Comment by warunalakshitha** [comment]
Lets merge this once https://github.com/ballerina-platform/ballerina-lang/pull/21093 merged

File: misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/TestAnnotationProcessor.java (hunk 1 above)
Comment (Dilhasha) [reply misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/TestAnnotationProcessor.java]: No we don't check that flag anywhere else. I'll combine the conditions.

**Comment by shafreenAnfar** [comment]
Closing this for the time being. Can reopen when needed.

**Pull request closed without merging**